jwst/master_background/master_background_step.py

```diff
@@ -90,7 +90,6 @@ def process(self, input):
                     del _
                     result = ModelContainer()
                     background_2d_collection = ModelContainer()
-                    background_2d_collection.update(input_data)
```

Comment:
Is it desirable to have association metadata, e.g. model.meta.asn.table_name and similar, in the model `background_2d_collection` when it's saved as an intermediate file?  If so, this needs to be replaced instead of just removed.  But I'm guessing it's probably not very important to retain those fields

Reply:
Oh, that hadn't occurred to me.  I doubt it's terribly important, since these products are diagnostic only.  Checking in the products now, though, it looks like that information is retained anyway because the intermediate products are made by copying the input products.  Is there anything that would be missed that way?

Reply:
no, in that case it should be fine.  The only thing I'd check would be `meta.asn.exptype`, although I imagine the call to `.update` would not have set that correctly to "background" in the first place.

Reply:
I'm not sure that field is appropriate to copy anyway. The masterbg2d products correspond to the science inputs, not the background members.  You could maybe argue that the masterbg1d product should get `meta.asn.exptype` 'background' -- it currently gets 'science'. But it's really a whole separate composite product that has never been included in an ASN, so it seems like a gray area.  Unless you disagree, I'll leave it as is.

But thanks either way for pointing this out!  I'll keep it in mind working with containers in the future.

Reply:
right, that's what I meant - it probably shouldn't get directly copied